packages/tools/src/utilities/math/ellipse/pointInEllipse.ts

```diff
@@ -36,7 +53,16 @@ export default function pointInEllipse(
     inverts.invYRadiusSq = yRadius !== 0 ? 1 / yRadius ** 2 : 0;
     inverts.invZRadiusSq = zRadius !== 0 ? 1 / zRadius ** 2 : 0;
   }
+  return inverts;
+};
 
+/**
+ * If you call the pointInEllipse.precalculateInverts first, then you
+ * can call precalculated directionly instead of having the extra time for
+ * the if conditions.
+ */
+pointInEllipse.precalculated = (ellipse, pointLPS, inverts) => {
```

Comment:
Nothing is wrong with that, but I wanted to optimize the time for that version since you were finding it slow.  This is probably the wrong optimization - the real one is to find the boundaries and iterate over the others in a simple loop.